pkg/compiler/transformers/k8s/workflow.go

```diff
@@ -191,6 +191,11 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 		interruptible = wf.GetMetadataDefaults().GetInterruptible()
 	}
 
+	var architecture core.Container_Architecture
+	for _, t := range tasks {
```

Comment:
this looks a bit weird. We just pick up the flag for the last task ?

Reply:
I agree that's weird.  I can update it to set the architecture only if it's set in one of the tasks and then break, which would cause all the tasks to run on the same architecture, but that would be necessary anyway, right? Tasks don't run on their own nodes, they all run within the same node, correct?